spec/support/shared_examples/abilities.rb

```diff
@@ -0,0 +1,95 @@
+# collections of shared examples
```

Comment:
Using shared examples is definitely a valid approach ✔️ 
I'm personally not a huge fan of using them in this amount and spread across multiple files though.

Well, comes down to personal preference, but I'm just quickly dumping my brain here: From my experience, they are usually only maintainable if being used only a few times and defined in the same file - it's the only way how a human brain can still keep track of the implicit contracts between the files. Plus: updating the feature and respectively the specs behind it gets super complicated, especially when nesting them like this.
They don't even run faster 😉 

Alternative approaches here could for instance be:
- split the abilities specs in one file per role _(prefer that tbh)_
- test based on the requested entity rather then the role _(and maybe again split into muliple files)_

Reply:
I totally agree that is not an ideal solution. It crossed my mind to split the abilities themselves in different files, but your suggestion to split only the specs in different files sounds better. I'd rather save that for a next step, if that is okay. 

Moreover, the shared examples are mostly dealing with 'authentication stages', not roles. If everything works out with the Big Overhaul, we don't have to have that in Ability anymore... 🍀

Reply:
Sure!

As said, your approach is totally valid 👍 

I'd personally have done it differently and just wanted to explain this in case you were actually looking for alternative approaches or did not think about certain gotchas that come with it.

So all good to stay with this.